File: stock-serveur-v2/src/main/java/cm/belrose/stockserveur/service/ArticleService.java
package cm.belrose.stockserveur.service;

import cm.belrose.stockserveur.dto.ArticleDto;
import cm.belrose.stockserveur.dto.LigneCommandeClientDto;
import cm.belrose.stockserveur.dto.LigneCommandeFournisseurDto;
import cm.belrose.stockserveur.dto.LigneVenteDto;
import cm.belrose.stockserveur.model.Article;

import java.util.List;

public interface ArticleService {
    ArticleDto save(ArticleDto dto);

    ArticleDto findById(Long id);

    ArticleDto findArticleByCode(String code);

    List<ArticleDto> findAll();

    List<LigneVenteDto> findHistoriqueVentes(Long idArticle);

    List<LigneCommandeClientDto> findHistoriqueCommandeClient(Long idArticle);

    List<LigneCommandeFournisseurDto> findHistoriqueCommandeFournisseur(Long idArticle);

    List<ArticleDto> findAllArticleByIdCategorie(Long idCategorie);


    void delete(Long id);
}

File: stock-client-v2/src/app/services/article/article.service.ts
import { Injectable } from '@angular/core';
import { Api2articlesService } from 'src/gs-api/src/services';
import { UserService } from '../user/user.service';
import { ArticleDto } from '../../../gs-api/src/models/article-dto';
import { Observable, of } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ArticleService {

  constructor(
    private userService: UserService,
    private api2ArticleService:Api2articlesService
  ) { }

  saveArticle(articleDto: ArticleDto): Observable<ArticleDto>{
    articleDto.entrepriseId = this.userService.getConnectedUser().entreprise?.id;
    return this.api2ArticleService.save(articleDto);

  }

  findAllArticle(): Observable<ArticleDto[]>{
    return this.api2ArticleService.findAll();
  }

  findArticleById(idArticle?: number): Observable<ArticleDto>{
    if (idArticle) {
      return this.api2ArticleService.findById(idArticle);
    }
    return of();
  }

  deleteArticle(idArticle: number): Observable<any>{
    if (idArticle) {
      return this.api2ArticleService.delete(idArticle);
    }
    return of();
  }
}
